pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/service/ActivityInstanceService.java

```diff
@@ -486,8 +486,10 @@ public void renderInstanceSummaries(Handle handle, long userId, String operatorG
             substitutions = substitutionStream
                     .collect(Collectors.toMap(SubstitutionsWrapper::getActivityInstanceId, SubstitutionsWrapper::unwrap));
         }
+
+        //TODO: Not sure if it is correct behaviour
         var sharedSnapshot = I18nContentRenderer
-                .newValueProviderBuilder(handle, userId, operatorGuid, studyGuid)
+                .newValueProviderBuilder(handle, instanceIds.iterator().next(), userId, operatorGuid, studyGuid)
```

Comment:
This is the weak place in my code. What instance ID should we use when the code works for multiple instances and creates a shared snapshot? I believe that any of them?

Reply:
Can we implement the needed functionality without saving data as snapshot? firstCompletedAt value for activity should always be available (once activity is completed).

Reply:
Should be easy enough, but it may have a performance impact since the `I18nContentRenderer` will be regenerated for _each_ activity summary (since there is no guarantee that all the activities here are related). I do not know if the current procedure is a case of premature optimization or not.